aws/data_source_aws_iam_role.go

```diff
@@ -78,10 +88,40 @@ func dataSourceAwsIAMRoleRead(d *schema.ResourceData, meta interface{}) error {
 	}
 	d.SetId(id)
 
-	data := resourceAwsIamRoleRead(d, meta)
+	input := &iam.GetRoleInput{
+		RoleName: aws.String(d.Id()),
+	}
+
+	output, err := iamconn.GetRole(input)
+	if err != nil {
+		return fmt.Errorf("Error reading IAM Role %s: %s", d.Id(), err)
+	}
+
+	d.Set("arn", output.Role.Arn)
+	if err := d.Set("create_date", output.Role.CreateDate.Format(time.RFC3339)); err != nil {
+		return err
+	}
+	d.Set("description", output.Role.Description)
+	d.Set("max_session_duration", output.Role.MaxSessionDuration)
+	d.Set("name", output.Role.RoleName)
+	d.Set("path", output.Role.Path)
+	d.Set("permissions_boundary", "")
```

Comment:
Do we have to set this to an empty string? I would think that Terraform would automatically do this but maybe I'm wrong.

Reply:
Since the `d.Set()` is wrapped in a conditional, this is to properly handle the missing attribute if its not sent back by the API (`output.Role.PermissionsBoundary` is `nil`). Otherwise Terraform will throw an error about the missing attribute in this scenario:

```hcl
output "permissions_boundary" {
  value = "${data.aws_iam_role.example.permissions_boundary}"
}
```

It can potentially go away after Terraform 0.12 allows returning full resources.